hooks/conf_regen/53-swap

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+source /usr/share/yunohost/helpers
+
+do_pre_regen() {
+	:
+}
+
+do_post_regen() {
+    swapfile_enabled="$(yunohost settings get 'swap.swapfile.swapfile_enabled')"
+    swapfile_size="$(yunohost settings get 'swap.swapfile.swapfile_size')"
+
+    if [ ${swapfile_enabled} -eq 1 ]; then
+        # If a swapfile is requested
+        if [ -f /swap_file ] && [ $(stat -c%s /swap_file) -ne $swapfile_size ]; then
+            # Let's delete it first if it is not of the requested size
+            ynh_print_info --message="Deleting swap first..."
+            ynh_del_swap
```

Comment:
That's really dangerous. Let's imagine this situation:
* RAM is 1GB, usage 900MB, Swapfile is 1GB, usage 500MB
* you want a 2GB swap file now, so we fall into this code
* ynh_del_swap deletes the original swap file, making the server (or services) most probably crash from OOM (1.4GB required on a 1GB available ram)

Reply:
Two better solutions would be :
* just create a second swap file of size = difference (in my example, 2GB(total) - 1GB(existing) = 1GB new swap file)
* create a second swap file of the total requested swap size, then afterwards delete the old file. The danger being the disk space but eh, i think we are okay, ram/swap is way smaller than disks usually…

Reply:
Ah good, thank you for giving some thoughts into this. :)
I had added a very lazy "Beware, if you reduce its current size, you may crash your server." in the config panel, but indeed there are other ways to crash everything.
I think your proposal is good!